tree/treeplayer/src/TTreeReader.cxx

```diff
@@ -380,6 +380,23 @@ bool TTreeReader::SetProxies() {
    return true;
 }
 
+void TTreeReader::WarnIfFriendsHaveMoreEntries()
+{
+   for (auto &&fp : fFriendProxies) {
```

Comment:
There should be no warning if the friend are indexed.  For example we can not tell whether all the entries in the main tree are using the same single entry in the friend (e.g. the friend contains information about luminosity block and all the entries in the main tree are from the same luminosity block).

Reply:
Ok thanks I will add the extra check here